django/contrib/postgres/forms/ranges.py

```diff
@@ -22,10 +22,17 @@ def __init__(self, **kwargs):
         super(BaseRangeField, self).__init__(**kwargs)
 
     def prepare_value(self, value):
+        lower_base, upper_base = self.fields
         if isinstance(value, self.range_type):
-            return [value.lower, value.upper]
+            return [
+                lower_base.prepare_value(value.lower),
+                upper_base.prepare_value(value.upper)
+            ]
         if value is None:
-            return [None, None]
+            return [
```

Comment:
Did you imagine a use case for preparing None? I guess the new test won't fail if it's reverted?